Source/Task/AsyncQueue.cpp

```diff
@@ -534,6 +534,12 @@ STDAPI CreateNestedAsyncQueue(
     _In_ async_queue_handle_t parentQueue,
     _Out_ async_queue_handle_t* queue)
 {
+    if (parentQueue == nullptr)
+    {
+        *queue = nullptr;
+        return S_OK;
+    }
```

Comment:
Should this just be in the if check below? Instead of return E_INVALIDARG

Reply:
? This looks like an error.  Why would I legitimately call this with a null parent?  I'd be tempted to crash.  Note that if you call with the queue field in providerData, that is never null (we put a default version in there if null was passed into the async block).